src/main/scala/com/microsoft/azure/iot/iothub2kafka/IotHubSourceTask.scala

```diff
@@ -98,36 +107,22 @@ class IotHubSourceTask extends SourceTask with LazyLogging with JsonSerializatio
     None
   }
 
-  private def getSavedOffsets(partitions: Iterable[String]): mutable.Map[String, String] = {
-
-    val partitionOffsetMap = mutable.Map.empty[String, String]
-    this.logger.debug("Getting offsets from previous values, if available")
-    if (this.context != null) {
-
-      val storageReader = this.context.offsetStorageReader()
-      if (storageReader != null && partitions != null) {
+  def getSavedOffset(offsetStorageReader: OffsetStorageReader, sourcePartition: util.Map[String, String]):
+  Option[String] = {
 
-        for (partition <- partitions) {
-
-          logger.debug(s"Getting saved offset for partition $partition")
-          val partitionKey = Collections.singletonMap("EventHubPartitionKey", partition)
-          val offsetMap = storageReader.offset(partitionKey)
-          if (offsetMap != null) {
-
-            if (offsetMap.containsKey("EventHubOffset")) {
-
-              offsetMap.get("EventHubOffset") match {
-                case storedOffset: String if storedOffset != null && storedOffset.trim.length > 0 =>
-                  partitionOffsetMap(partition) = storedOffset.trim
-                  this.logger.info(s"Setting offset from previous snapshot for partition $partition " +
-                    s"to value - $storedOffset")
-              }
-            }
-          }
+    if (offsetStorageReader != null) {
+      logger.debug(s"Attempting to get saved offset for SourcePartition: ${sourcePartition.toString}")
+      val offsetMap = offsetStorageReader.offset(sourcePartition)
+      if (offsetMap != null && offsetMap.containsKey("EventHubOffset")) {
+        offsetMap.get("EventHubOffset") match {
```

Comment:
In the other cases, I am not keying off the same variable in the if else, so will keep it as it is for now.